### ISSUE TITLE
Better initial guesses in ExampleDsl.NlpMonad

### ISSUE THREAD

**Issue by aavogt** [ghorn/dynobud#74]
http://code.haskell.org/~aavogt/ipopt-hs/examples/WilliamsOtto_casadi.hs needs one variable (T) specified as an initial guess to avoid making NaNs.

A saner guess could come from looking at the constraints `5.8 < T < 6.8`. Both williamsOtto.ampl and WilliamsOtto.hs there solve the problem without an explicit initial guess.

**Comment by ghorn** [comment]
I don't quite understand the suggestion. Are you suggesting that the example DSL should take initial guesses from looking at the bounds?

**Comment by aavogt** [comment]
exactly: if you don't specify an initial guess I think the initial guess actually used should satisfy the bounds
